### PR TITLE
[MDEP-788] Upgrade maven-reporting-impl to version 3.1.0

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- use maven-site-plugin in IT test
- remove dependency to doxia, it is transitive from maven-reporting-impl

---

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MDEP) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MDEP-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MDEP-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the integration tests successfully (`mvn -Prun-its clean verify`).

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

File: src/it/projects/analyze-report/pom.xml [hunk 1]
```diff
@@ -59,13 +59,40 @@
...
-          <configuration>
-           <outputXML>true</outputXML>
-           <scriptableOutput>true</scriptableOutput>
-           <verbose>true</verbose>
-          </configuration>
+          <version>@project.version@</version>
+        </plugin>
+        <plugin>
+          <artifactId>maven-site-plugin</artifactId>
+          <version>3.10.0</version>
```
Comment (michael-o) [inline src/it/projects/analyze-report/pom.xml:66]: 3.11.0 is in the works which upgrades the entire stack. I recommend to wait for it.

Comment (elharo) [reply src/it/projects/analyze-report/pom.xml:66]: IMHO not a good idea. It's not worth delaying this, and it's no harder to update to 3.11.0 before or after this PR.

**Review by elharo — APPROVED** [review]

File: src/it/projects/analyze-report/pom.xml (hunk 1 above)
Comment (michael-o) [reply src/it/projects/analyze-report/pom.xml:66]: Three days more or less, who cares?

Comment (slawekjaranowski) [reply src/it/projects/analyze-report/pom.xml:66]: This is IT test ... project also override m-site-plugin from parent
 
In such situation will be good to have some possibility to inherit plugin version from parent pom, in other case IT test will have outdated dependency in some time and will be not maintain

So I prefer to update m-site-p in separate PR

**Pull request merged**